fed/proxy/barriers.py

```diff
@@ -235,15 +202,33 @@ async def send(
         dest_addr = self._cluster[dest_party]['address']
         dest_party_grpc_config = self.setup_grpc_config(dest_party)
         try:
+            tls_enabled = fed_utils.tls_enabled(self._tls_config)
+            grpc_options = dest_party_grpc_config['grpc_options']
+            grpc_options = get_grpc_options(retry_policy=self.retry_policy) if \
+                grpc_options is None else fed_utils.dict2tuple(grpc_options)
+
+            if dest_party not in self._stubs:
+                if tls_enabled:
+                    ca_cert, private_key, cert_chain = fed_utils.load_cert_config(
+                        self._tls_config)
+                    credentials = grpc.ssl_channel_credentials(
+                        certificate_chain=cert_chain,
+                        private_key=private_key,
+                        root_certificates=ca_cert,
+                    )
+                    channel = grpc.aio.secure_channel(
+                        dest_addr, credentials, options=grpc_options)
+                else:
+                    channel = grpc.aio.insecure_channel(dest_addr, options=grpc_options)
+                stub = fed_pb2_grpc.GrpcServiceStub(channel)
```

Comment:
Maybe we should close the channel explicitly on shutdown?

Reply:
All about this are  in the actor, which will be destroyed when `fed.shutdown`.
so I don't think it's a big deal.